lldb/source/ValueObject/ValueObject.cpp

```diff
@@ -734,15 +734,22 @@ size_t ValueObject::GetPointeeData(DataExtractor &data, uint32_t item_idx,
     } break;
     case eAddressTypeLoad: {
       ExecutionContext exe_ctx(GetExecutionContextRef());
-      Process *process = exe_ctx.GetProcessPtr();
-      if (process) {
-        heap_buf_ptr->SetByteSize(bytes);
-        size_t bytes_read = process->ReadMemory(
+      heap_buf_ptr->SetByteSize(bytes);
+      size_t bytes_read = 0;
+      if (Process *process = exe_ctx.GetProcessPtr();
+          process && process->IsLiveDebugSession()) {
```

Comment:
I'd like to avoid special casing core files. Can we just call the target version all the time?